x-pack/plugins/uptime/server/rest_api/pings/journey_screenshots.ts

```diff
@@ -42,3 +45,44 @@ export const createJourneyScreenshotRoute: UMRestApiRouteFactory = (libs: UMServ
     });
   },
 });
+
+export const createLastSuccessfulStepScreenshotRoute: UMRestApiRouteFactory = (
+  libs: UMServerLibs
+) => ({
+  method: 'GET',
+  path: '/api/uptime/step/screenshot/{monitorId}/{stepIndex}',
+  validate: {
+    params: schema.object({
+      monitorId: schema.string(),
+      stepIndex: schema.number(),
+    }),
+    query: schema.object({
+      timestamp: schema.string(),
+      _debug: schema.maybe(schema.boolean()),
+    }),
+  },
+  handler: async ({ uptimeEsClient, request, response }) => {
+    const { monitorId, stepIndex } = request.params;
+    const { timestamp } = request.query;
+
+    const result = await libs.requests.getStepLastSuccessfulScreenshot({
+      uptimeEsClient,
+      monitorId,
+      stepIndex,
+      timestamp,
+    });
+
+    if (result === null) {
+      return response.notFound();
+    }
+    return response.ok({
+      body: Buffer.from(result.blob, 'base64'),
+      headers: {
+        'content-type': 'image/png',
+        'cache-control': 'max-age=600',
+        'caption-name': result.stepName,
```

Comment:
Why are we returning information as headers?

Reply:
this is refactored into separate API